sovrin_client/agent/endpoint.py

```diff
@@ -62,8 +63,10 @@ def baseMsgHandler(self, msg):
     def connectTo(self, ha):
         remote = self.findInRemotesByHA(ha)
         if not remote:
-            remote = RemoteEstate(stack=self, ha=ha)
-            self.addRemote(remote)
-            # updates the store time so the join timer is accurate
-            self.updateStamp()
-            self.join(uid=remote.uid, cascade=True, timeout=30)
+            while True:
+                name = randomString(8)
+                if name in self.registry: continue
+                else: break
+
+            self.registry[name] = ha
+            self.connect(name)
```

Comment:
It's tough to see from the code here if it actually fixes a problem. We need at least one test of this functionality that fails before the fix and works after the fix.